mysite/lti/views.py

```diff
@@ -101,21 +118,72 @@ def lti_redirect(request, course_id=None, unit_id=None):
 
     if not user.is_linked:
         user.create_links()
-
     user.login(request)
+
+    if not course_id or not Course.objects.filter(id=course_id).exists():
+        if course_ref:
+            course_id = course_ref.course.id
+        elif Role.INSTRUCTOR in roles:
+            return redirect(reverse('lti:create_courseref'))
+        else:
+            return render_to_response(
+                'lti/error.html',
+                {'message': """You are trying to access Course that does not exists but
+                            Students can not create new Courses automatically"""},
+                RequestContext(request)
+            )
+
     user.enroll(roles, course_id)
 
     if user.is_enrolled(roles, course_id):
         # Redirect to course or unit page considering users role
         if not unit_id:
             dispatch = 'ct:course_student'
-            if roles == 'prof':
+            if Role.INSTRUCTOR in roles:
                 dispatch = 'ct:course'
             return redirect(reverse(dispatch, args=(course_id,)))
         else:
             dispatch = 'ct:study_unit'
-            if roles == 'prof':
+            if Role.INSTRUCTOR in roles:
                 dispatch = 'ct:unit_tasks'
             return redirect(reverse(dispatch, args=(course_id, unit_id)))
     else:
         return redirect(reverse('ct:home'))
+
+
+@only_lti
+def create_courseref(request):
+    """
+    Create CourseRef and Course entry based on context_title.
+    """
+    request_dict = pickle.loads(request.session['LTI_POST'])
+    if not request.session.get('is_valid'):
+        return redirect(reverse('ct:home'))
+    context_id = request_dict.get('context_id')
+    roles_from_request = request_dict.get('roles', '').split(',')
+    roles = list(set((ROLES_MAP.get(role, Role.ENROLLED) for role in roles_from_request)))
+    # Make sure this context_id is not used
+    course_ref = CourseRef.objects.filter(context_id=context_id).first()
+    if course_ref:
+        if Role.INSTRUCTOR in roles:
+            return redirect(reverse('ct:edit_course', args=(course_ref.course.id,)))
```

Comment:
Actually, I'm not sure that what Instructor wants in this scenario is ct:edit_course.  Maybe just ct:course would be better